com.unity.render-pipelines.universal/ShaderLibrary/Shadows.hlsl

```diff
@@ -350,17 +350,17 @@ half GetMainLightShadowFade(float3 positionWS)
     float3 camToPixel = positionWS - _WorldSpaceCameraPos;
     float distanceCamToPixel2 = dot(camToPixel, camToPixel);
 
-    half fade = saturate(distanceCamToPixel2 * _MainLightShadowParams.z + _MainLightShadowParams.w);
-    return fade;
+    float fade = saturate(distanceCamToPixel2 * float(_MainLightShadowParams.z) + float(_MainLightShadowParams.w));
```

Comment:
is _MainLightShadowParams using half?

Reply:
Yes, it's defined: `half4 _MainLightShadowParams;` in Shadows.hlsl.